modules/cnf-migrating-from-amqp-to-http-transport.adoc

```diff
@@ -0,0 +1,184 @@
+
+.Prerequisites
+
+* You have installed the OpenShift CLI (`oc`).
+
+* You have logged in as a user with `cluster-admin` privileges.
+
+* You have updated the PTP Operator or {redfish-operator} to version 4.13+ which uses HTTP transport by default.
+
+.Procedure
+
+. Configure persistent storage to persist events subscriptions in the pod managed by the PTP Operator or {redfish-operator}.
+For example, apply the following `PersistentVolumeClaim` (PVC) custom resource (CR) in the cluster:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: cloud-event-proxy-store
+spec:
+  storageClassName: local-storage
+  resources:
+    requests:
+      storage: 10Mi
+  accessModes:
+  - ReadWriteOnce
+----
+
+. Configure the PTP events producer application with a corresponding `persistentVolumeClaim` that matches the PVC that you have configured in step 1. For example:
++
+[source,yaml]
+----
+volumes:
+  - name: pubsubstore
+    persistentVolumeClaim:
+      claimName: cloud-event-proxy-store
+----
+
+. Get the name of the PTP or bare-metal events `transportHost` in the cluster. For example:
+
+.. To get the name of the PTP events `transportHost`, run the following command:
++
+[source,terminal]
+----
+$ oc get ptpoperatorconfig -n openshift-ptp -o jsonpath='{ .items[*].spec.ptpEventConfig.transportHost }'
+----
++
+.Example output
+[source,terminal]
+----
+http://ptp-event-publisher-service-compute-1-example.openshift-ptp.svc.cluster.local:9043
+----
+
+.. To get the name of the bare-metal events `transportHost`, run the following command:
++
+[source,terminal]
+----
+$ oc -n openshift-bare-metal-events get hardwareevent -o jsonpath='{.items[*].spec.transportHost}{"\n"}'
+----
++
+.Example output
+[source,terminal]
+----
+http://hw-event-publisher-service.openshift-bare-metal-events.svc.cluster.local:9043
+----
+
+. Update your events consumer application to use HTTP transport.
+
+.. Set the `transport-type` variable in the cloud events consumer application deployment, for example:
++
+[source,yaml]
+----
+containers:
+  - name: cloud-event-consumer
+    image: <your_cloud_event_consumer_app_image>
+    args:
+      - "--local-api-addr=127.0.0.1:9089"
+      - "--api-path=/api/ocloudNotifications/v1/"
+      - "--api-addr=127.0.0.1:8089"
+      - "--transport-type=HTTP"
+----
+
+.. Set the `http-event-publishers` variable for the cloud event sidecar deployment to match the value that you set for `transportHost` in the cluster `PtpOperatorConfig` CR, for example:
++
+[source,yaml]
+----
+containers:
+- name: cloud-event-sidecar
+  image: registry.redhat.io/openshift4/ose-cloud-event-proxy
+  args:
+    - "--metrics-addr=127.0.0.1:9091"
+    - "--store-path=/store"
+    - "--transport-host=consumer-events-subscription-service.cloud-events.svc.cluster.local:9043"
```

Comment:
It's worth to mention a `consumer-events-subscription-service` service needs to be deployed at consumer side to support the http transport.

https://github.com/redhat-cne/cloud-event-proxy/blob/main/examples/manifests/service.yaml
```
apiVersion: v1
kind: Service
metadata:
  annotations:
    prometheus.io/scrape: "true"
    service.alpha.openshift.io/serving-cert-secret-name: sidecar-consumer-secret
  name: consumer-events-subscription-service
  namespace: cloud-events
  labels:
    app: consumer-service
spec:
  ports:
    - name: sub-port
      port: 9043
  selector:
    app: consumer
  clusterIP: None
  sessionAffinity: None
  type: ClusterIP
```

Reply:
Should this service also be added as an additional step here: https://55695--docspreview.netlify.app/openshift-enterprise/latest/networking/using-ptp.html#cnf-configuring-the-ptp-fast-event-publisher_using-ptp

Reply:
yes